osu.Game/Skinning/LegacySkin.cs

```diff
@@ -311,17 +305,27 @@ public override Texture GetTexture(string componentName)
 
         public override SampleChannel GetSample(ISampleInfo sampleInfo)
         {
-            foreach (var lookup in sampleInfo.LookupNames)
+            var hsi = sampleInfo as HitSampleInfo;
+
+            if (hsi?.IsLayered == true && GetConfig<LegacySkinConfiguration.LegacySetting, bool>(LegacySkinConfiguration.LegacySetting.LayeredHitSounds)?.Value == false)
```

Comment:
This isn't being read from the skin configuration. You need to add a value to the `LegacySetting` switch above.